lib/discourse_api/api/topics.rb

```diff
@@ -85,6 +85,13 @@ def change_owner(topic_id, params = {})
 
         post("/t/#{topic_id}/change-owner.json", params)
       end
+
+      def topic_set_user_notification(args = {})
```

Comment:
For url params, let's just pass them in as method args so `def topic_set_user_notification(category_id, args = {})`
and then we don't need the `category_id = args[:id]` line.

Just noticed that you followed the pattern from `category_set_user_notification`, but let's not follow that pattern.

Reply:
I had the same thought. Think now is a good time to refactor `category_set_user_notification` as well? I can make that update while I'm in there.

Reply:
Yes we should fix it if you are up for it, but what we should do is deprecate that method and make a new one so that we don't break things right away for people who may be using it. See https://github.com/discourse/discourse_api/blob/99446adaa476d11f058552440199f47d962d66f8/lib/discourse_api/api/topics.rb#L50 for an example of how to do that.

Reply:
@mikepack Hi just following up and seeing if you are still up for updating this PR? Would love to get these changes merged in so we can cut another release.